src/Logging.AzureAppServices/src/BlobLoggerProvider.cs

```diff
@@ -58,10 +58,19 @@ public BlobLoggerProvider(IOptionsMonitor<AzureBlobLoggerOptions> options)
         internal override async Task WriteMessagesAsync(IEnumerable<LogMessage> messages, CancellationToken cancellationToken)
         {
             var eventGroups = messages.GroupBy(GetBlobKey);
+            DateTime currDate = DateTime.Now;
             foreach (var eventGroup in eventGroups)
             {
                 var key = eventGroup.Key;
-                var blobName = $"{_appName}/{key.Year}/{key.Month:00}/{key.Day:00}/{key.Hour:00}/{_fileName}";
+                string blobName;
+
+                if (!string.IsNullOrEmpty(AzureAppServicesLoggerFactoryExtensions.CustomPrefix))
```

Comment:
This shouldn't be looked up via a static, but instead should be an instance.

Reply:
This entire Azure Blob / File Log API is only called once in the beginning of the program run. I could create a new singleton class that contains this one string, but I believe that would be overkill. Unless that is the preferred approach? Please let me know. 
I have tested this solution in Azure and it works flawlessly both ways. The original way, where the blob is structured under appname/yyyy/mm/dd/hh/appInstanceId_blobname, and the modified way, where the blob is named {customprefix}{yyyy}(mm){dd} .

The only difference in setup is whether or not to pass the customPrefix name

ihost.ConfigureLogging(x =>
{
x.AddAzureWebAppDiagnostics(**customPrefixName**);
});

or

ihost.ConfigureLogging(x =>
{
x.AddAzureWebAppDiagnostics**()**;
});